### DISCUSSION TITLE
Should addon repositories only contain type definitions?

### DISCUSSION THREAD

**Discussion by jonasgeiler** [LuaLS/LLS-Addons#134] [General]
I'm currently working on a library called [lua-fenster](https://github.com/jonasgeiler/lua-fenster) (still experimental) and want to provide type definitions for it. Is it intended for me to create a separate repository using the [addon template](https://github.com/LuaLS/addon-template) or can I just reuse the repository I already have where the actual library resides, and just add a `config.json` and `library` folder there?
I would like to keep everything together in one place for easier contribution and maintenance. I suppose that type definition repos should be kept minimal since they get cloned by the addon manager every time, right? Or is it okay to just clone the whole library along with the definitions?

**Comment by carsakiller** [comment]
The addon manager was designed to clone a small repository containing just the definitions and what is necessary. You *can* just include all the implementation code as well, but it will likely make the addon much larger and slower to parse.

**Reply by jonasgeiler** [reply]
In my case there won't be any implementation code in `library/`, only the plain type definitions. I would expect parsing to be as fast as any other library except for the part where it downloads the repository since that includes the actual implementation code in C which is like 300KB max - nothing crazy.
